src/org/mozilla/javascript/TokenStream.java

```diff
@@ -1185,6 +1206,62 @@ final int getToken() throws IOException
         }
     }
 
+    /*
+     * Helper to read the next digits according to the base
+     * and ignore the number separator if there is one.
+     */
+    private int readDigits(int base, int c) throws IOException {
+        if ((base == 10 && isDigit(c))
+                || (base == 16 && isHexDigit(c))
+                || (base == 8 && isOctalDigit(c))
+                || (base == 2 && isDualDigit(c))) {
```

Comment:
Minor, but I see that we have repeated this four-line conditional three times in this code. Would you consider DRYing this and moving it into a separate private function?

Reply:
done